packages/@aws-cdk/aws-appmesh/test/test.virtual-gateway.ts

```diff
@@ -153,6 +155,151 @@ export = {
+          }),
+        })],
+      });
+
+      // THEN
+      expect(stack).to(haveResourceLike('AWS::AppMesh::VirtualGateway', {
+        Spec: {
+          Listeners: [
+            {
+              PortMapping: {
+                Port: 8080,
+                Protocol: appmesh.Protocol.HTTP,
+              },
+              TLS: {
+                Mode: TlsMode.STRICT,
+                Certificate: {
+                  ACM: {
+                    CertificateArn: {
+                      Ref: 'cert56CA94EB',
+                    },
+                  },
+                },
+              },
+            },
+          ],
+        },
+      }));
+
+      test.done();
+    },
+    'with an grpc listener with a TLS certificate from file'(test: Test) {
+      // GIVEN
+      const stack = new cdk.Stack();
+
+      // WHEN
+      const mesh = new appmesh.Mesh(stack, 'mesh', {
+        meshName: 'test-mesh',
+      });
+
+      new appmesh.VirtualGateway(stack, 'testGateway', {
+        virtualGatewayName: 'test-gateway',
+        mesh: mesh,
+        listeners: [appmesh.VirtualGatewayListener.grpc({
+          port: 8080,
+          tlsCertificate: appmesh.TlsCertificate.file({
+            certificateChain: 'path/to/certChain',
+            privateKey: 'path/to/privateKey',
+            tlsMode: TlsMode.STRICT,
+          }),
+        })],
+      });
+
+      // THEN
+      expect(stack).to(haveResourceLike('AWS::AppMesh::VirtualGateway', {
+        Spec: {
+          Listeners: [
+            {
+              PortMapping: {
+                Port: 8080,
+                Protocol: appmesh.Protocol.GRPC,
+              },
+              TLS: {
+                Mode: TlsMode.STRICT,
+                Certificate: {
+                  File: {
+                    CertificateChain: 'path/to/certChain',
+                    PrivateKey: 'path/to/privateKey',
+                  },
+                },
+              },
+            },
+          ],
+        },
+      }));
+
+      test.done();
+    },
+    'with an grpc listener with the TLS mode permissive'(test: Test) {
+      // GIVEN
+      const stack = new cdk.Stack();
+
+      // WHEN
+      const mesh = new appmesh.Mesh(stack, 'mesh', {
+        meshName: 'test-mesh',
+      });
+
+      new appmesh.VirtualGateway(stack, 'testGateway', {
+        virtualGatewayName: 'test-gateway',
+        mesh: mesh,
+        listeners: [appmesh.VirtualGatewayListener.grpc({
+          port: 8080,
+          tlsCertificate: appmesh.TlsCertificate.file({
+            certificateChain: 'path/to/certChain',
+            privateKey: 'path/to/privateKey',
+            tlsMode: TlsMode.PERMISSIVE,
+          }),
+        })],
+      });
+
+      // THEN
+      expect(stack).to(haveResourceLike('AWS::AppMesh::VirtualGateway', {
+        Spec: {
+          Listeners: [
+            {
+              PortMapping: {
+                Port: 8080,
+                Protocol: appmesh.Protocol.GRPC,
```

Comment:
I would remove this part from all the tests - they're verified in different ones. Make these only focus on the part of the resource they're interested in (in this case, the 'TLS' part).

Reply:
ah Ididn't look into `haveResourceLike(...)` I assumed these were exact match. cool!